nwb_conversion_tools/utils/basesinwbephyswriter.py

```diff
@@ -33,8 +33,7 @@ def _get_unit_sampling_frequency(self):
 
     @default_return([])
     def _get_unit_ids(self):
-        if self.sorting is not None:
-            return self.sorting.get_unit_ids()
+        return self.sorting.get_unit_ids()
```

Comment:
even with the decorator, if `self.sorting is None` this will fail because it would try to get `self.sorting.get_unit_ids()`

Reply:
the decorator also checks if there is an AttributeError with the 'NoneType' in its comments. Basically, if we try to look up an attribute for a None object. It will still return the default return value then. [This Line](https://github.com/catalystneuro/nwb-conversion-tools/pull/230/files/b6823f957472ffe478bd2d64036a8010149d68db#diff-1b7da2ff140e7bc2daa681d1fdcf7458864e4080decec994675cc95d261df5acR123)

Reply:
the decorator also checks if we're calling some method/attribute on a NoneType (if self.recording/sorting etc are None). Then it also returns the default value also.